socketio/src/asynchronous/client/client.rs

```diff
@@ -148,11 +233,13 @@ impl Client {
     /// }
     /// ```
     pub async fn disconnect(&self) -> Result<()> {
+        self.manually_disconnected.store(true, Ordering::Relaxed);
```

Comment:
I think this should be `Ordering::Release`? We want to synchronize with every read.

Reply:
Fixed it